test/distributed/fsdp/test_fsdp_comm_hooks.py

```diff
@@ -123,6 +167,8 @@ def test_default_communication_hook_behavior(
 
             # For each worker, the gradient on the weight should be worker_rank.
             grad = net_default_hook.params[0].grad
+            if sharding_strategy != ShardingStrategy.NO_SHARD:
+                self.assertTrue(net_default_hook.params[0]._is_sharded, "Expected gradient to be a sharded chunk")
```

Comment:
I do not think the assertion message matches what is being asserted.

The message describes the gradient, but the assertion does not test the gradient.